pkg/services/notary/core_test.go

```diff
@@ -600,22 +600,14 @@ func TestNotary(t *testing.T) {
 	// Add block before allowing tx to finalize to exclude race condition when
 	// main transaction is finalized between `finalizeWithError` restore and adding new block.
 	e.AddNewBlock(t)
-	mtx.RLock()
-	start := len(completedTxes)
-	mtx.RUnlock()
 	setFinalizeWithError(false)
-	for i := range requests {
-		if i != 0 {
-			e.AddNewBlock(t)
-		}
-		require.Eventually(t, func() bool {
-			mtx.RLock()
-			defer mtx.RUnlock()
-			return len(completedTxes)-start >= i+1
-		}, time.Second*3, time.Millisecond)
+	for i := range requests[1:] {
+		e.AddNewBlock(t)
+
+		_ = getCompletedTx(t, true, requests[i+1].FallbackTransaction.Hash())
```

Comment:
Good fix, so we wait for `i+1` fallback to be completed, and after that check that no other fallbacks were completed, right?

While we're here, could you please adjust the `getCompletedTx` comment, since we use it not only for main transactions, but also for fallbacks:
```
}, time.Second*3, time.Millisecond*50, errors.New("main transaction expected to be completed"))
```

Reply:
Yes, fallback 0 can be or can be not added, but once we add one more block --- it's there for sure.

Fixed.